client/client.go

```diff
@@ -717,13 +785,98 @@ func newWebsocketDialerTLS(conf Config) (*websocket.Dialer, error) {
 
 	dialer := websocket.Dialer{
 		NetDialTLSContext: func(_ context.Context, network string, addr string) (net.Conn, error) {
+			proxyUrl, err := getProxyUrlFromHostPort(addr)
+			if err != nil {
+				log.Errorf("Failed to get http-proxy configurations during websocket dial: %v", err)
+			} else if proxyUrl != nil {
+				return proxyDialOpenSSL(ctx, &conf, network, addr, proxyUrl)
+			}
 			return dialOpenSSL(ctx, &conf, network, addr)
 		},
 	}
 
 	return &dialer, nil
 }
 
+func getProxyUrlFromHostPort(addr string) (*url.URL, error) {
+	u, err := url.Parse("https://" + addr)
+	if err != nil {
+		return u, err
+	}
+	// presumably for security reasons, http.ProxyFromEnvironment returns
+	// nil if req.URL is localhost. To aid testing, allow localhost proxying
+	// only if both HTTPS_PROXY and req.URL are locahosts
```

Comment:
We should not do this. We need to find a way for the tests to override this method without hard-coding "test aide" code in the routine. We can have a public method like `client.ProxxyUrlFromHostPortGetter` in the package set to the safe function and for the tests to be able to override it with this special tweaking.